File: vs/src/CodeStream.VisualStudio/Models/Agent.Protocol.Scm.cs
using Microsoft.VisualStudio.LanguageServer.Protocol;
using System.Collections.Generic;

namespace CodeStream.VisualStudio.Models
{
    public class GetRangeScmInfoRequest
    {
        public string Uri { get; set; }
        public Range Range { get; set; }
        public bool Dirty { get; set; }
        public string Contents { get; set; }
    }

    public class Author
    {
        public string Id { get; set; }
        public string Username { get; set; }
    }

    public class Remote
    {
        public string Name { get; set; }
        public string Url { get; set; }
    }

    public class Scm
    {
        public string File { get; set; }
        public string RepoPath { get; set; }
        public string Revision { get; set; }
        public List<Author> Authors { get; set; }
        public List<Remote> Remotes { get; set; }
    }

    public class GetRangeScmInfoResponse
    {
        public string Uri { get; set; }
        public Range Range { get; set; }
        public string Contents { get; set; }
        public Scm Scm { get; set; }
        public string Error { get; set; }
    }

    public class GetRangeScmInfoRequestType : RequestType<GetRangeScmInfoRequest>
    {
        public static string MethodName = "codestream/scm/range/info";
        public override string Method => MethodName;
    }

}

File: jb/src/main/kotlin/protocols/agent/AgentProtocolScm.kt
package protocols.agent

import org.eclipse.lsp4j.Range

interface GetRangeScmInfoRequest {
    val uri: String
    val range: Range
    val dirty: Boolean
    val contents: String?
}

interface Author {
    val id: String
    val username: String
}

interface Remote {
    val name: String
    val url: String
}

interface Scm {
    val file: String
    val repoPath: String
    val revision: String
    val authors: Array<Author>
    val remotes: Array<Remote>
}

interface GetRangeScmInfoResponse {
    val uri: String
    val range: Range
    val contents: String
    val scm: Scm?
    val error: String?
}